content/en/blog/_posts/2021-07-16-read-write-once-pod-access-mode-alpha.md

```diff
@@ -0,0 +1,280 @@
+
+## How do I use it?
+
+The ReadWriteOncePod access mode is in alpha for Kubernetes v1.22 and is only supported for CSI volumes.
+As a first step you need to enable the ReadWriteOncePod [feature gate](/docs/reference/command-line-tools-reference/feature-gates) for `kube-apiserver`, `kube-scheduler`, and `kubelet`.
+You can enable (or disable) the feature by setting command line arguments:
+
+```
+--feature-gates="...,ReadWriteOncePod=true"
+```
+
+### Creating a PersistentVolumeClaim
+
+In order to use the ReadWriteOncePod access mode for your PVs and PVCs, you will need to create a new PVC with the access mode:
+
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: single-writer-only
+spec:
+  accessModes:
+  - ReadWriteOncePod # Allow only a single pod to access single-writer-only.
+  resources:
+    requests:
+      storage: 1Gi
+```
+
+If your storage plugin supports [dynamic provisioning](/docs/concepts/storage/dynamic-provisioning/), new PersistentVolumes will be created with the ReadWriteOncePod access mode applied.
+
+#### Migrating existing PersistentVolumes
+
+If you have existing PersistentVolumes, they can be migrated to use ReadWriteOncePod.
+
+In this example, we already have a "cat-pictures-pvc" PersistentVolumeClaim that is bound to a "cat-pictures-pv" PersistentVolume, and a "cat-pictures-writer" Deployment that uses this PersistentVolumeClaim.
+
+As a first step, you need to edit your PersistentVolume's `spec.persistentVolumeReclaimPolicy` and set it to `Retain`.
+This ensures your PersistentVolume will not be deleted when we delete the corresponding PersistentVolumeClaim:
+
+```shell
+kubectl patch pv cat-pictures-pv -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}'
+```
+
+Next you need to stop any workloads that are using the PersistentVolumeClaim bound to the PersistentVolume you want to migrate, and then delete the PersistentVolumeClaim.
+
+Once that is done, you need to clear your PersistentVolume's `spec.claimRef.uid` to ensure PersistentVolumeClaims can bind to it upon recreation:
+
+```shell
+kubectl scale --replicas=0 deployment cat-pictures-writer
+kubectl delete pvc cat-pictures-pvc
+kubectl patch pv cat-pictures-pv -p '{"spec":{"claimRef":{"uid":""}}}'
+```
+
+After that you need to replace the PersistentVolume's access modes with ReadWriteOncePod:
+
+```shell
+kubectl patch pv cat-pictures-pv -p '{"spec":{"accessModes":["ReadWriteOncePod"]}}'
+```
+
+{{< note >}}
+The ReadWriteOncePod access mode cannot be combined with other access modes.
+Make sure ReadWriteOncePod is the only access mode on the PersistentVolume when updating, otherwise the request will fail.
+{{< /note >}}
+
+Next you need to modify your PersistentVolumeClaim to set ReadWriteOncePod as the only access mode.
+You should also set your PersistentVolumeClaim's `spec.volumeName` to the name of your PersistentVolume.
+Doing this guarantees your PersistentVolumeClaim will bind to this specific PersistentVolume.
```

Comment:
It's actually the opposite (setting PV.spec.claimref) that guarantees the binding. If you only set the PVC.spec.volumeName, you could run into an issue where multiple PVCs set the same PV.